src/render/tutorials/chrome-plugin/2_2.html.md

```diff
@@ -0,0 +1,495 @@
+
+Let's start with the following task:
+
+```hocon
+user-task {
+  action = fetch-user-info
+}
+```
+
+We defined `user-task` as an object that represents a root node of the graph. Right now, this node contains a single property `action` that defines what action this node is supposed to call. Think of actions as behaviours. 
+
+In this tutorial, we are using mocks with JSON data, so fetching a user should call an appropriate endpoint via HTTP. This is the configuration for the `fetch-user-info` action:
+
+```hocon
+fetch-user-info {
+  factory = http
+  config.endpointOptions {
+    path = /user
+    domain = webapi
+    port = 8080
+  }
+}
+```
+
+The most important property to explain is the `factory` piece. This defines which java factory should produce the action. Think of it as of a type of the action. In this case we use the `http` factory (a factory registered under the name "http"). It's a built-in action that calls a specified address and retrieves data from it. The `config.endpointOptions` property should be self explanatory. In case you're wondering why the domain is defined as `webapi` - this example runs as a docker swarm configured so that mocks are available at this host. If you're not using docker, it'd be localhost.
+
+Note that you can also write and register your own factories using java, but it's not in the scope of this tutorial.
+
+If we were to leave the configuration like this we would create a REST API. We want to render a page though, so we need another node in our task. When the first node succeeds (so user data is fetched), we want to call a node that renders a page. This is the updated code for the `user-task`:
+
+```hocon
+user-task {
+  action = fetch-user-info
+  on._success {
+    action = te-hbs
+  }
+}
+```
+
+The `on` object contains all transitions (graph edges) we want to be able to handle. The format looks like this:
+
+```hocon
+on {
+  transition1 = node1
+  transition2 = node2
+  // ...
+}
+```
+
+In this case we define that after a node with action `fetch-user-info` Knot.x should call a node with action `te-hbs`. This action will render the page using handlebars. This is how it looks like:
+
+```hocon
+te-hbs {
+  factory = knot
+  config.address = knotx.knot.te.handlebars
+}
+```
+
+Perhabs not what you were expecting. The `knot` factory allows for registering actions on the [Vert.x Event Bus](https://vertx.io/docs/vertx-core/java/#event_bus), thus providing more scalability. Here's how the actual handlebars configuration looks like:
+
+```hocon
+address = knotx.knot.te.handlebars

```

Comment:
Maybe we should provide some scalability options here:
```
modules {
  hbsTemplateEngine = "io.knotx.te.core.TemplateEngineKnot"
}
config.hbsTemplateEngine.options {
  instances = 4
  config {
    address = knotx.knot.te.handlebars
    engine {
      factory = handlebars
      config = {
        cacheSize = 1000
      }
    }
  }
}
```
?

Reply:
Knot action (`te-hbs`) delegates the processing to "workers" that listens on a particular address (`knotx.knot.te.handlebars`). With that, we can fully utilise CPU resources (horizontal scaling) or use cluster mode to deploy modules vertically (on different machines) not affecting graph simplicity.